casanovo/denovo/model_runner.py

```diff
@@ -23,307 +25,350 @@
-        n_layers=config["n_layers"],
-        dropout=config["dropout"],
-        dim_intensity=config["dim_intensity"],
-        custom_encoder=config["custom_encoder"],
-        max_length=config["max_length"],
-        residues=config["residues"],
-        max_charge=config["max_charge"],
-        precursor_mass_tol=config["precursor_mass_tol"],
-        isotope_error_range=config["isotope_error_range"],
-        n_beams=config["n_beams"],
-        top_match=config["top_match"],
-        n_log=config["n_log"],
-        tb_summarywriter=config["tb_summarywriter"],
-        warmup_iters=config["warmup_iters"],
-        max_iters=config["max_iters"],
-        lr=config["learning_rate"],
-        weight_decay=config["weight_decay"],
-    )
-    if config["train_from_scratch"]:
-        model = Spec2Pep(**model_params)
-    else:
-        if not os.path.isfile(model_filename):
+        if train and any(from_scratch):
+            self.model = Spec2Pep(**model_params)
+            return
+        elif self.model_filename is None:
+            logger.error("A model file must be provided")
+            raise ValueError("A model file must be provided")
+
+        if not Path(self.model_filename).exists():
             logger.error(
-                "Could not find the model weights at file %s to continue "
-                "training",
-                model_filename,
-            )
-            raise FileNotFoundError(
-                "Could not find the model weights to continue training"
-            )
-        model = Spec2Pep().load_from_checkpoint(model_filename, **model_params)
-    # Create the Trainer object and (optionally) a checkpoint callback to
-    # periodically save the model.
-    if config["save_model"]:
-        callbacks = [
-            pl.callbacks.ModelCheckpoint(
-                dirpath=config["model_save_folder_path"],
-                save_top_k=-1,
-                save_weights_only=config["save_weights_only"],
-                every_n_train_steps=config["every_n_train_steps"],
+                "Could not find the model weights at file %s",
+                self.model_filename,
             )
-        ]
-    else:
-        callbacks = None
-
-    trainer = pl.Trainer(
-        accelerator="auto",
-        auto_select_gpus=True,
-        callbacks=callbacks,
-        devices=_get_devices(config["no_gpu"]),
-        enable_checkpointing=config["save_model"],
-        logger=config["logger"],
-        max_epochs=config["max_epochs"],
-        num_sanity_val_steps=config["num_sanity_val_steps"],
-        strategy=_get_strategy(),
-        val_check_interval=config["every_n_train_steps"],
-    )
-    # Train the model.
-    trainer.fit(
-        model, train_loader.train_dataloader(), val_loader.val_dataloader()
-    )
-    # Clean up temporary files.
-    tmp_dir.cleanup()
+            raise FileNotFoundError("Could not find the model weights file")
+
+        self.model = Spec2Pep().load_from_checkpoint(
```

Comment:
@melihyilmaz What does this FIXME mean?
> # FIXME: Handle the case for mps separately.
Can we just address it rather than leave this as an open todo?

Reply:
Also, does the GitHub CI provide different devices to write a unit test for this issue, or is it CPU-only? [This](https://github.com/pytorch/pytorch/issues/61654) might be useful.

Reply:
re: MPS case, I couldn't find how to obtain from the `Trainer` whether an mps accelerator is being used to pass that on to `map_location`

Reply:
re: unit testing devices, I think the weight loading tests are disabled now but with weights matching the architecture, this case will automatically be tested as gpu-trained weights would be loaded into the model which is run on a cpu.

Reply:
Why not just always load to CPU, then transfer to whatever device for training?

Reply:
@wfondrie Meaning we set `map_location_device = "cpu"` by default and then move the model weights to whatever device in `trainer`? (if so some quick googling didn't yield how to do the latter)

Reply:
If this leads to an error, I do think it's good to have an explicit unit test for it, rather than have it implied by the weights loading test. Good suggestion by @wfondrie.